Submitted By: Jim Gifford (jim at linuxfromscratch dot org)
Date: 2004-02-08
Initial Package Version: 2.01
Origin: http://kerneltrap.org/node/view/879
Description: Fixes CDRTools Compile with the 2.6 Kernel
 
diff -Naur cdrtools-2.01.orig/libscg/scsi-linux-sg.c cdrtools-2.01/libscg/scsi-linux-sg.c
--- cdrtools-2.01.orig/libscg/scsi-linux-sg.c	2004-01-14 17:54:01.000000000 +0000
+++ cdrtools-2.01/libscg/scsi-linux-sg.c	2004-02-08 19:15:06.179407934 +0000
@@ -65,6 +65,13 @@
 
 #if LINUX_VERSION_CODE >= 0x01031a /* <linux/scsi.h> introduced in 1.3.26 */
 #if LINUX_VERSION_CODE >= 0x020000 /* <scsi/scsi.h> introduced somewhere. */
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
+	#define __KERNEL__
+	#include <asm/types.h>
+	#undef __KERNEL__
+#endif
+
 /* Need to fine tune the ifdef so we get the transition point right. */
 #include <scsi/scsi.h>
 #else
